### PR TITLE
BugFix for issue 13482

### PR DESCRIPTION
**Pull request by nickwinger**
https://github.com/protocolbuffers/protobuf/issues/13482

### CODE REVIEW

**Comment by nickwinger** [comment]
@jskeet , @tonydnewell here is the PR for the Bug

File: src/google/protobuf/compiler/csharp/names.cc [hunk 1]
```diff
@@ -184,6 +184,15 @@ std::string UnderscoresToCamelCase(absl::string_view input,
     } else if ('0' <= input[i] && input[i] <= '9') {
       result += input[i];
       cap_next_letter = true;
+    // https://github.com/protocolbuffers/protobuf/issues/13482
+    // Preserve underscores before numbers if the come right after a dot
+    } else if (
```
Comment (jskeet) [inline src/google/protobuf/compiler/csharp/names.cc:189]: Drive-by comment - I haven't analysed this fully - what would this do with "my.scope.__3"? I *suspect* we'd still end up with "my.scope.3", which presumably isn't intended.

Comment (nickwinger) [reply src/google/protobuf/compiler/csharp/names.cc:189]: yes you are right, there is room for improvement, however my fix doesn't break anything and it still is an improvement. double underscore handling should be another PR as it makes things even more complicated. however most of the time when people only use one underscore it is an improvement
